src/hooks/useCumulativeFundingRates.js

```diff
@@ -0,0 +1,40 @@
+import { useWeb3React } from '@web3-react/core';
+import { ethers } from 'ethers';
+import React, { useEffect } from 'react';
+import { getContract } from '../Addresses';
+import { RPC_PROVIDERS } from '../Helpers';
+import Vault from '../abis/Vault.json';
+
+
+export const useCumulativeFundingRates = (chainId, nativeTokenAddress, whitelistedTokenAddresses) => {
+    const [cumulativeFundingRates, setCumulativeFundingRates] = React.useState(null);
+    const vaultAddress = getContract(chainId, "Vault");
+
+
+    useEffect(() => {
+        handleFetchCumulativeFundingRates();
+    }, [chainId]);
+    
+    async function handleFetchCumulativeFundingRates() {
+        try {
+            const provider = new ethers.providers.JsonRpcProvider(RPC_PROVIDERS[chainId][0]);
+            await provider.ready;
+            const vault = new ethers.Contract(vaultAddress, Vault.abi, provider);
```

Comment:
do you think we should use { library } and the useSWR design pattern? that includes the fallback provider and stuff